cli/cli.cc

```diff
@@ -245,12 +245,14 @@ int tr_main(int argc, char* argv[])
 
         if (!tr_sys_path_exists(sz_download_dir))
         {
-            tr_error* error = nullptr;
-
-            if (!tr_sys_dir_create(sz_download_dir, TR_SYS_DIR_CREATE_PARENTS, 0700, &error))
+            if (auto error = tr_error{}; !tr_sys_dir_create(sz_download_dir, TR_SYS_DIR_CREATE_PARENTS, 0700, &error) && error)
             {
-                fprintf(stderr, "Unable to create download directory \"%s\": %s\n", sz_download_dir.c_str(), error->message);
-                tr_error_free(error);
+                auto const errmsg = fmt::format(
+                    "Couldn't create '{path}': {error} ({error_code})",
+                    fmt::arg("path", sz_download_dir),
+                    fmt::arg("error", error.message()),
+                    fmt::arg("error_code", error.code()));
+                fmt::print(stderr, "{:s}\n", errmsg);
```

Comment:
This suggested change loses the linefeed.

We could just add it to the end of the string, but I'd prefer not to because `"Couldn't create '{path}': {error} ({error_code})"` is already in several other places and my hope is to (eventually) unify all the apps' translatable strings.

I agree with your thought process (i.e. "only use this string in one place, why not skip the middle step") but the alternate version reads more like:

```diff
diff --git a/cli/cli.cc b/cli/cli.cc
index 621d1abf9..c3ff96861 100644
--- a/cli/cli.cc
+++ b/cli/cli.cc
@@ -247,12 +247,14 @@ int tr_main(int argc, char* argv[])
         {
             if (auto error = tr_error{}; !tr_sys_dir_create(sz_download_dir, TR_SYS_DIR_CREATE_PARENTS, 0700, &error) && error)
             {
-                auto const errmsg = fmt::format(
-                    "Couldn't create '{path}': {error} ({error_code})",
-                    fmt::arg("path", sz_download_dir),
-                    fmt::arg("error", error.message()),
-                    fmt::arg("error_code", error.code()));
-                fmt::print(stderr, "{:s}\n", errmsg);
+                fmt::print(
+                    stderr,
+                    "{:s}\n",
+                    fmt::format(
+                        "Couldn't create '{path}': {error} ({error_code})",
+                        fmt::arg("path", sz_download_dir),
+                        fmt::arg("error", error.message()),
+                        fmt::arg("error_code", error.code())));
                 return EXIT_FAILURE;
             }
         }
```

This is a subjective call, but IMO this is uglier than the version in the PR

Reply:
Fair enough 👍🏻